src/realm/object-store/sync/sync_manager.hpp

```diff
@@ -75,7 +75,10 @@ struct SyncClientConfig {
 
     std::string base_file_path;
     MetadataMode metadata_mode = MetadataMode::Encryption;
-    util::Optional<std::vector<char>> custom_encryption_key;
+    std::optional<std::vector<char>> custom_encryption_key;
+#if REALM_PLATFORM_APPLE
+    std::string security_access_group;
+#endif
```

Comment:
Rather than conditionally enable this only for apple platforms, would it make sense to either throw or ignore this on non-apple platforms? The reason is that most of the cross-platform SDKs ship a platform-agnostic library which exposes the same API across all target OS-es. While we could check and throw further up the stack and conditionally call this like you do in the C API, it's a little bit of extra code that every SDK will need to write and probably also spend some time on aligning on the behavior we want.

Reply:
Yeah, probably no reason to make this conditional.

Reply:
I think it doesn't even specifically need a runtime check on non-Apple platforms. It's only applicable when encryption is enabled but no key is set, which will already throw an error.